File: ROS/ROS_Lab2/cpp_lab2/src/node1.cpp
// 1) include ros client library and message types 
#include <chrono>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "cpp_lab2/msg/myinfo.hpp"


using namespace std::chrono_literals;

// 2) Create class that inherit from rcl::Node calss
class Node1 : public rclcpp::Node
{
  public:
// 4) Inside class constructor define( node name(uniqe), publisher , timer )
    Node1(): Node("minimal_publisher")
    { 
      //create publisher object this->create_publisher<msg type>("topic_name",queue_size)
      publisher_ = this->create_publisher<cpp_lab2::msg::Myinfo>("topic2", 10);
      // create timer to control publishing rate 
      timer_ = this->create_wall_timer(1000ms, std::bind(&Node1::timer_callback, this));
    }

//3)Inside private declare (Publisher , Subscriber , Timer , Counter , callbacks , services ,..)
  private:
    // 1) define callback function
    void timer_callback()
    {
      //step(1):create message object
      auto message = cpp_lab2::msg::Myinfo();
      //step(2):fill message members
      message.data = "Mohamed Abdelwahab is publishing: " + std::to_string(count_++);
      message.num = count_ ;
      // print statement
      RCLCPP_INFO(this->get_logger(), "Mohamed Abdelwahab is publishing:  '%d'", message.num);
      //step(3):publish the message
      publisher_->publish(message);
    }

    // 2)create timer pointer / placeholder
    rclcpp::TimerBase::SharedPtr timer_;
    // 3)create publisher pointer / placeholder
    rclcpp::Publisher<cpp_lab2::msg::Myinfo>::SharedPtr publisher_;
    size_t count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Node1>());
  rclcpp::shutdown();
  return 0;
}

File: ROS/ROS_Lab3/iti_lab3/iti_lab3/node1.py
#!/usr/bin/env python3

import rclpy

from rclpy.node import Node
from my_msgs.msg import Newms


#inherit from Node
class my_node(Node):
    def __init__(self):
        super().__init__("pub_node")
        self.create_timer(.5,self.callback)
        self.obj_pub = self.create_publisher(Newms,"new_topic",10)
        self.get_logger().info("node is started now")
        self.counter = 0

    def callback(self):
        #self.get_logger().info("Mohamed Abdelwahab 12345")
        msg = Newms()
        msg.message = "Mohamed Abdelwahab is publishing : 5 "
        msg.num = 5
        
        
        self.obj_pub.publish(msg)




def main(args=None):
    rclpy.init(args=args)

    node = my_node()
    rclpy.spin(node)
    



    rclpy.shutdown()


if __name__ == "__main__":
    main()






